### PR TITLE
Feature/auth-app/UT-203

### PR DESCRIPTION
**Pull request by icaroharry**
### Verify that
- [x] the branch name follows the git flow naming convention (feature/i18n)
- [x] the UT reference is appended to the branch name if it concerns a Jira Task (feature/i18n/UT-283)
- [x] the design is responsive
- [x] the code doesn't break Server Side Rendering
- [ ] new functions are prefixed with JSDoc information
- [ ] the pull request contains unit tests for new code that passes and has at least 50% coverage
- [x] the code doesn't break the api endpoints contract
- [x] JS Standard code style has been verified with ESLint

Implements [UT-203](https://utopian-io.atlassian.net/browse/UT-203)

This is how the auth app works:
* Separate quasar app 
* `yarn dev` runs the auth app on `localhost:8081`
* Redirect the user to `localhost:8081/login?redirectUrl=myurl`
* The login page will show up and after the user connect his GitHub account he will be redirected to `redirectUrl`

### **Test plan**
* Run the app with `yarn dev` on the root folder
* Verify that `@utopian/api` is running on port 5000, `@utopian/client` is running on port 8080 and `@utopian/auth` is running on port 8081

_**Scenario 1** User doesn't have account_

* Go to localhost:8080
* Click on the `Sign in` button in the toolbar
* You will be redirected to `localhost:8081/login`
* Click on the `Sign in with GitHub` button (translation missing)
* You will be redirected to `/users/create`
* Create a new account

**Expected result:** You should be redirected to the main app page with a signed in user. Verify that the `access_token` and the `refresh_token` are set in the store and in the cookies.

_**Scenario 2** User has an account_

* Go to localhost:8080
* Click on the `Sign in` button in the toolbar
* You will be redirected to `localhost:8081/login`
* Click on the `Sign in with GitHub` button (translation missing)

**Expected result:** You should be redirected to the main app page with a signed in user. Verify that the `access_token` and the `refresh_token` are set in the store and in the cookies.

### CODE REVIEW

File: packages/auth/src/pages/redirect/redirect.vue [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<script>
+export default {
+  preFetch ({ store, redirect }) {
+    if (store.state.auth.user) {
+      window.location = process.env.UTOPIAN_DOMAIN
```
Comment (gregory-latinier) [inline packages/auth/src/pages/redirect/redirect.vue]: The preFetch happens on the server side.
window is undefined and will crash the app

File: packages/auth/quasar.conf.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ module.exports = function (ctx) {
     build: {
       env: {
         UTOPIAN_API: process.env.UTOPIAN_API,
+        AUTH_DOMAIN: process.env.AUTH_DOMAIN,
```
Comment (gregory-latinier) [inline packages/auth/quasar.conf.js]: Can you update the .env files ?

File: packages/auth/src/css/utopian/colors.styl [hunk 3]
```diff
@@ -0,0 +1,47 @@
+$dusk = #3e3e65
```
Comment (gregory-latinier) [inline packages/auth/src/css/utopian/colors.styl]: Why the styles are still there?

File: packages/auth/src/pages/redirect/redirect.vue (hunk 1 above)
Comment (icaroharry) [reply packages/auth/src/pages/redirect/redirect.vue]: removed this function. It's not necessary anymore

File: packages/auth/quasar.conf.js (hunk 2 above)
Comment (icaroharry) [reply packages/auth/quasar.conf.js]: done

File: packages/auth/src/css/utopian/colors.styl (hunk 3 above)
Comment (icaroharry) [reply packages/auth/src/css/utopian/colors.styl]: sorry. they were accidentally committed again

File: packages/auth/src/plugins/auth/index.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  // Link blockchain accounts
+  if (loginState === 'steemconnectlogin') {
+    await linkSteemAccount({ currentRoute, store, redirectUrl })
+  }
+
+  // Load the user information
+  if (cookies.get('access_token')) {
+    const token = jwt.decode(cookies.get('access_token'))
+    if (token.scopes.includes('app')) {
+      await store.dispatch('auth/me')
```
Comment (gregory-latinier) [inline packages/auth/src/plugins/auth/index.js]: You don't need the user information in this app

File: packages/auth/src/store/auth/actions.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { Cookies } from 'quasar'
+import API from 'src/plugins/api'
+
+export const me = async (context) => {
```
Comment (gregory-latinier) [inline packages/auth/src/store/auth/actions.js]: This is not needed

File: packages/auth/src/store/auth/getters.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { get } from 'lodash'
+
+export const guest = ({ user }) => (user === null)
```
Comment (gregory-latinier) [inline packages/auth/src/store/auth/getters.js]: Can you clean up the getters?

File: packages/auth/src/store/auth/mutations.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+export const setUser = (state, user = null) => {
```
Comment (gregory-latinier) [inline packages/auth/src/store/auth/mutations.js]: Clean up the mutations as well

File: packages/auth/src/store/auth/state.js [hunk 8]
```diff
@@ -0,0 +1,4 @@
+export default () => ({
```
Comment (gregory-latinier) [inline packages/auth/src/store/auth/state.js]: No need for this either

File: packages/auth/src/store/users/actions.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    context,
+    method: 'get',
+    url: `/v1/user/${username}/available`
+  })
+
+  return payload.available
+}
+
+export const saveUser = async (context, data) => {
+  console.log(Cookies, process.env.SERVER)
```
Comment (gregory-latinier) [inline packages/auth/src/store/users/actions.js]: Boouhh :D

File: packages/auth/src/store/users/state.js [hunk 10]
```diff
@@ -0,0 +1,3 @@
+export default () => ({
```
Comment (gregory-latinier) [inline packages/auth/src/store/users/state.js]: Is it needed?

File: packages/auth/src/css/themes/shadows.styl [hunk 11]
```diff
@@ -0,0 +1,52 @@
+$shadow-0 = 0 0 0 $elevation-umbra, 0 0 0 $elevation-penumbra, 0 0 0 $elevation-ambient
```
Comment (gregory-latinier) [inline packages/auth/src/css/themes/shadows.styl]: Is it needed ?

File: packages/auth/src/pages/redirect/redirect.vue [hunk 12]
```diff
@@ -0,0 +1,16 @@
+<script>
+export default {
```
Comment (gregory-latinier) [inline packages/auth/src/pages/redirect/redirect.vue]: Just a question. Why do you need this page? Can't you redirect directly without using this page?

**Comment by gregory-latinier** [comment]
The authentication works fine that's great

File: packages/auth/src/store/auth/getters.js [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import { get } from 'lodash'
```
Comment (nothingismagick) [inline packages/auth/src/store/auth/getters.js]: Why are you still using lodash?

File: packages/auth/src/store/users/actions.js (hunk 9 above)
Comment (nothingismagick) [reply packages/auth/src/store/users/actions.js]: Coming soon to a test suite near you:
```
console.log = jest.fn(() => { throw new Error('Do not use console.log()') })
```

File: packages/client/.env.example [hunk 14]
```diff
@@ -1,3 +1,4 @@
 UTOPIAN_API='"http://localhost:5000"'
 GITHUB_CLIENT_ID='""'
 STEEMCONNECT_CLIENT_ID='""'
+AUTH_DOMAIN='"http://localhost:8081"'
```
Comment (nothingismagick) [inline packages/client/.env.example]: missing empty line at the end...

File: packages/client/src/components/layout/toolbar/toolbar.pug [hunk 15]
```diff
@@ -5,7 +5,7 @@ q-toolbar.u-toolbar(color='tertiary')
       img.u-logo.desktop-only(src="~assets/img/logo-white.svg")
 
     div.u-toolbar-right(v-if="guest === true")
-      q-btn(@click.native="startGithubLogin", color="primary", icon="mdi-github-circle", label="Login with GitHub")
+      q-btn(@click.native="redirectToLogin", color="primary", icon="mdi-account", label="Sign in")
```
Comment (nothingismagick) [inline packages/client/src/components/layout/toolbar/toolbar.pug]: label should be:
```
:label="$t(common.signIn)"
```
or something

File: packages/client/src/pages/login/login.pug [hunk 16]
```diff
@@ -8,4 +8,4 @@ q-layout.u-page-login
           p.text-center.subtitle {{$t('login.subtitle')}}
           q-btn.q-mt-sm.signup(@click="startGithubLogin", icon="mdi-github-circle", color="primary", :label="$t('login.button.signUp')")
       div.side-img.col-md-8
-        img(src="~assets/img/background.jpg")
+        img(src="~assets/img/background.jpeg")
```
Comment (nothingismagick) [inline packages/client/src/pages/login/login.pug]: **.jpeg** ??? this is not 1991.

File: packages/client/src/store/users/actions.js [hunk 17]
```diff
@@ -9,33 +8,4 @@ export const isUsernameAvailable = async (context, username) => {
...
-    path: '/',
-    expires: 365
-  })
-  Cookies.set('access_token', payload.tokens.access_token, {
-    path: '/',
-    expires: 365
-  })
-  return payload
-}
+}
```
Comment (nothingismagick) [inline packages/client/src/store/users/actions.js]: Always put a blank line at the end of files.

File: packages/auth/src/store/auth/getters.js (hunk 6 above)
Comment (icaroharry) [reply packages/auth/src/store/auth/getters.js]: ok. I'm leaving just the steem related ones that might be needed for the next step of the auth app

**Pull request merged**